src/core/server/services/comments/comments.ts

```diff
@@ -79,3 +109,175 @@ export async function removeTag(
 
   return removeCommentTag(mongo, tenant.id, commentID, tagType);
 }
+
+export async function retrieveComment(
+  mongo: MongoContext,
+  tenantID: string,
+  id: string,
+  skipArchive?: boolean
+) {
+  const liveComment = await retrieveCommentModel(
+    mongo.comments(),
+    tenantID,
+    id
+  );
+
+  if (liveComment) {
+    return liveComment;
+  }
+
+  const archivedComments = mongo.archivedComments();
+  if (mongo.archive && !skipArchive && archivedComments) {
+    const archivedComment = await retrieveCommentModel(
+      archivedComments,
+      tenantID,
+      id
+    );
+
+    return archivedComment;
+  }
+
+  return null;
+}
+
+export async function retrieveManyComments(
+  mongo: MongoContext,
+  tenantID: string,
+  ids: ReadonlyArray<string>
+) {
+  if (ids.length === 0) {
+    return [];
+  }
+
+  const liveComments = await retrieveManyCommentModels(
+    mongo.comments(),
+    tenantID,
+    ids
+  );
+  if (liveComments.length > 0) {
```

Comment:
Should filter out null elements, before this check. (I know it's confusing as hell)

Reply:
Sth like `liveComments.filter(c => Boolean(c)).length > 0`